### ISSUE TITLE
🚀 Feature: Postgres Support

### ISSUE THREAD

**Issue by shakks** [appwrite/appwrite#2541]
### 🔖 Feature description

Hi, i tried to look around to find any details regarding the plans to support Postgres, is there any plan to support it. If its already supported is there any documents. Thanks

### 🎤 Pitch

Our backend uses a Postgres DB.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by lohanidamodar** [comment]
Hi @shakks, Postgres is not yet supported, however we do plan to support Postgres. You will be able to see the progress in our [database library](https://github.com/utopia-php/database) when the work starts.

**Comment by caevv** [comment]
The only thing holding me back from moving to appwrite. Hopefully, this is still in progress.

**Comment by amirping** [comment]
any news? that's the only missing + for appwrite

**Comment by eqoram** [comment]
I personally think that more people would go with Appwrite when the underlying database would be Postgres (same goes for the cloud)! Would be great to have this option in the future (:

**Comment by jmealo** [comment]
I don't understand why new projects are using MariaDB. The real-time worker could have really benefited from Logical Decoding. You could keep the cache populated without hitting the DB. It also makes it easier to scale out real time/read-only nodes.  It made me kind of pick the rest of the project apart because I question the engineering decisions now. I see a lot of good, a lot of bad, but if ya'll had went with Postgres you'd probably have another contributor :)

**Comment by luke-stdev001** [comment]
any news on this front? We'd love to run AppWrite on top of our horizontally scaling postgres infrastructure as our expertise and tooling is all focused around Postgres.

**Comment by cahofmeyr** [comment]
I also saw that Utopia Database lists Postgres and therefore have the same question as @luke-stdev001

**Comment by marcbejar** [comment]
In the Utopia docs they say that they have already fully integrated a Postgres adapter into utopia library. As I understand, now it's time to wait until this library function is integrated into AppWrite... I hope they don't take too long, this is the main reason I'm still developing with Supabase. @cahofmeyr

**Comment by lostb1t** [comment]
+1 for this

**Comment by dishankjindal1** [comment]
Any update?

**Comment by nhdms** [comment]
Bump this update?

**Comment by rdr6000** [comment]
I am also waiting for an update on this.

**Comment by SaidBahaDev** [comment]
+1

**Comment by rickylabs** [comment]
@lohanidamodar any news on this topic ?
I sincerely hope they will announce PG support during _init event next week !

Long awaited features that could shake the concurrent (Supabase, Hasura, Nhost, ...)

**Comment by KingPin** [comment]
Yeah, Postgres support is a blocker for my project also. Would love to hear some updates on this please.

**Comment by Splinteer** [comment]
I think it's now supported:
https://github.com/utopia-php/database?tab=readme-ov-file#supported-databases

**Comment by KingPin** [comment]
> I think it's now supported: https://github.com/utopia-php/database?tab=readme-ov-file#supported-databases

yeah, I saw on the utopia library page, wasn't sure if something is needed on appwrite side to support it, and devs haven't responded to anyone in a while.

**Comment by sawa-ko** [comment]
I need this too!

**Comment by platform-kit** [comment]
@lohanidamodar can you shed any light on this?

**Comment by jamroks** [comment]
I'm also in need of PostgreSql Support. But i just saw in your public roadmap , there is nothing plan for Potsgres :-(

**Comment by onlyamessenger** [comment]
@jamroks Looks like they work according to amount of upvotes. The MongoDB feature included in version 1.7 has 53 upvotes. While this Postgres feature only has 39. If we can get more people to upvote this I think we'll be in a good spot to have this included in version 1.8.

**Comment by evanheckert** [comment]
Disconcerting that project devs haven't commented on this to give any sense of the roadmap.

**Comment by rickylabs** [comment]
@eldadfux, could you enlighten us about this topic?

**Comment by CordialGit** [comment]
My Application is a SvelteKit Application, while I want to use a Postgres database for backend.

Now, Appwrite uses MariaDB. 

I am of the opinion that Appwrite should have Postgres support, and give us the ability to select our Database at set up.

The reason for this is because-- suppose my Application grows in future, and I feel that I want to migrate my data from Appwrite, I want to be able to directly migrate all my database to my new Server's Postgres database, without downtime and without data loss.

+1

**Comment by rdlogout** [comment]
+1

**Comment by QThans** [comment]
+1

**Comment by renegadevi** [comment]
+1

**Comment by FaisalMohammadi** [comment]
+1

**Comment by AmiinYuusuf** [comment]
+1

**Comment by asynckebap** [comment]
+1

**Comment by polvallverdu** [comment]
+1

**Comment by stefanadelbert** [comment]
+1

**Comment by onrdmr** [comment]
+1

**Comment by Fractal-Tess** [comment]
+1

**Comment by zebra-pro-dev** [comment]
+1

**Comment by satputekuldip** [comment]
+1

**Comment by onlyamessenger** [comment]
@eldadfux This is the issue is currently number 6 in terms of upvote count. Any idea on when it will be added to the backlog? Or guidance on how the community can help?

**Comment by sam-morin** [comment]
+1

**Comment by sergioponguta** [comment]
+1

**Comment by Azreil-OFD** [comment]
+1

**Comment by terryhycheng** [comment]
+1

**Comment by Nathaniel-Fernandes** [comment]
+1

**Comment by Hanibachi** [comment]
three years since this issue was opened and the devs said it was soon to be integrated .... what exactly does "soon" mean in their dictionary.

**Comment by enrio** [comment]
+1

**Comment by fnrcum** [comment]
+1

**Comment by chirag-urb** [comment]
+1

**Comment by felixrydberg** [comment]
+1

**Comment by suitcodemr** [comment]
+1

**Comment by Infrns** [comment]
+1

**Comment by zebra-pro-dev** [comment]
+1

**Comment by gaozuo** [comment]
+1

**Comment by stevebrainng** [comment]
+

**Comment by onlyamessenger** [comment]
Just keep voting people :D

**Comment by ammarfaris** [comment]
+1

**Comment by j0ng4b** [comment]
+1

**Comment by luisotavio-dev** [comment]
+1

**Comment by onlyamessenger** [comment]
Just watching init announcements hoping for this

**Comment by gaozuo** [comment]
any news?

**Comment by ammarfaris** [comment]
> For those wanting news, I believe [this PR](https://github.com/appwrite/appwrite/pull/9772) is linked to this feature.

wow niceeeee

**Comment by felixrydberg** [comment]
I wonder if you will be able to use Neon 🤔

**Comment by neolizee** [comment]
+1

**Comment by xingchen-zh** [comment]
+1

**Comment by ryonsoft** [comment]
Any news on postgres support guys?

**Comment by onlyamessenger** [comment]
I think we should give up on this gents. I commented the question on one of their YouTube videos recently and they just deleted it.

**Comment by mrbyrk** [comment]
@onlyamessenger They are planning it for 1.8 release.

**Comment by onlyamessenger** [comment]
@mrbyrk Thanks for the update. Where did you see the items for the 1.8 release?

**Comment by xunjianxiang** [comment]
+1

**Comment by abnegate** [comment]
Interested to know, what are people's expectations for Postgres support? As Appwrite provides an abstraction layer, the common interactions with the database would remain the same - it would just be Postgres under the hood.

Are you looking for:
- Postgres specific features
- Raw SQL access
- Direct database connection

**Comment by rhengles** [comment]
@abnegate In my opinion it's more about robustness, reliability and performance instead of using some exclusive feature.

Also could be good for scalability, but that may require more code and configuration than simply connecting to Postgres.

**Comment by DH-555** [comment]
@rhengles Scalability isn't an advantage for Postgres support at all. 
MariaDB is already very performant and scalable. A proof is Appwrite cloud, that is handling thousands of projects and billions of monthly database operations.

**Comment by onlyamessenger** [comment]
@abnegate Postgres is just so feature-rich. If there is ever a need for lower-level access, Postgres offers more options.

**Comment by appurist** [comment]
@abnegate I'm looking for more direct input of large amounts of data (many records). I have about 8 hours of importing to do, every month, if one at a time.  It's more like 15 minutes if I could directly import data.

+1 for Postgres support, or AppWrite support for multiple-record import in a single function execution.
